nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -212,9 +215,21 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
         final Map<String, String> statePropertyMap = new HashMap<>(stateMap.toMap());
 
         //If an initial max value for column(s) has been specified using properties, and this column is not in the state manager, sync them to the state property map
-        for(final Map.Entry<String,String> maxProp : maxValueProperties.entrySet()){
-            if (!statePropertyMap.containsKey(maxProp.getKey().toLowerCase())) {
-                statePropertyMap.put(maxProp.getKey().toLowerCase(), maxProp.getValue());
+        for (final Map.Entry<String, String> maxProp : maxValueProperties.entrySet()) {
```

Comment:
When we add incoming flow file support, it maybe useful if Initial max values can be specified via flow file attributes. Because if table name is dynamic, user won't be able to configure dynamic processor attribute in advance for each possible table.

Example Flow File (table.name='table-1', maxvalue.columns='update_date', initial.maxvalue.update_date='20170101')